src/util/subscribeToResult.ts

```diff
@@ -51,9 +63,9 @@ export function subscribeToResult<T, R>(outerSubscriber: OuterSubscriber<T, R>,
       root.setTimeout(() => { throw err; });
     });
     return destination;
-  } else if (typeof result[$$iterator] === 'function') {
-    for (let item of result) {
-      destination.next(item);
+  } else if (isIterable(result)) {
```

Comment:
I can put them back, it was mainly helping with dealing with the typings, with the type guards. (revealed a few issues)